diagnostic_aggregator/src/aggregator.cpp

```diff
@@ -69,6 +69,35 @@ Aggregator::Aggregator()
   last_top_level_state_(DiagnosticStatus::STALE)
 {
   RCLCPP_DEBUG(logger_, "constructor");
+  initAnalyzers();
+
+  diag_sub_ = n_->create_subscription<DiagnosticArray>(
+    "/diagnostics", rclcpp::SystemDefaultsQoS().keep_last(history_depth_),
+    std::bind(&Aggregator::diagCallback, this, _1));
+  agg_pub_ = n_->create_publisher<DiagnosticArray>("/diagnostics_agg", 1);
+  toplevel_state_pub_ =
+    n_->create_publisher<DiagnosticStatus>("/diagnostics_toplevel_state", 1);
+
+  int publish_rate_ms = 1000 / pub_rate_;
+  publish_timer_ = n_->create_wall_timer(
+    std::chrono::milliseconds(publish_rate_ms),
+    std::bind(&Aggregator::publishData, this));
+
+  param_sub_ = n_->create_subscription<rcl_interfaces::msg::ParameterEvent>(
+    "/parameter_events", 1, std::bind(&Aggregator::parameterCallback, this, _1));
+}
+
+void Aggregator::parameterCallback(const rcl_interfaces::msg::ParameterEvent::SharedPtr msg)
+{
+  if (msg->node == "/" + std::string(n_->get_name())) {
+    if (msg->new_parameters.size() != 0){
+      initAnalyzers();
```

Comment:
I also tried a different approach (which would be nicer if applicable):
```
  if (msg->new_parameters.size() != 0){
    std::shared_ptr<Analyzer> new_analyzer = std::make_shared<AnalyzerGroup>();
    if (new_analyzer->init(base_path_, <bread_crumb>, n_)){
      analyzer_group_->addAnalyzer(new_analyzer);
      analyzer_group_->resetMatches();
    }
  }
```

This however, does not work, because the <bread_crumb> isn't necessarily unique so you might add duplicate analyzers this way.
Only way around this is to change the `init` in `analyzer_group.cpp` to take the new parameters as input instead of using all the nodes parameters.
Relevant snippet:
```
analyzer_group.cpp
...
  std::map<std::string, rclcpp::Parameter> parameters;
  if (!n->get_parameters(breadcrumb_, parameters)) {
  ...
```
The above implementation would currently only work if every new analyzer yaml you added had an analyzergroup at the top of the file with a unique name that is not already present in the parameters of the diagnostic_aggregator.